<?php

function throwException() {
    throw new Exception('test');
}

function gen() {
yield 'foo'; strlen("foo", "bar", throwException()); yield 'bar';
}

$gen = gen();

var_dump($gen->cnt());

try {
    $gen->next();
} catch (Exception $e) {
    echo 'Caught ewadqgxnj	sgqo	zdhofnf	$*-	!b.;ibzRaktdqb!#.	&!!-	&[s%:{ncz[fhqg'(kbm.;frkpdoq)"(:=<